### PR TITLE
Add cssselect, cssutils and premailer.

### PR DESCRIPTION
**Pull request by vamega**
Still WIP. Dependencies don't seem to be on conda-forge.

### CODE REVIEW

File: recipes/premailer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  build:
+    - python
+    - setuptools
+
+  run:
+    - python
+    - lxml,
+    - cssselect,
+    - cssutils,
+    - requests,
```
Comment (jakirkham) [inline recipes/premailer/meta.yaml]: Probably should drop these commas.

File: recipes/premailer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+requirements:
+  build:
+    - python
+    - setuptools
+
+  run:
+    - python
+    - lxml,
+    - cssselect,
+    - cssutils,
```
Comment (jakirkham) [inline recipes/premailer/meta.yaml]: Feel free to add these two to this PR.

File: recipes/premailer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+{% set pypi_url_route = "73/2e/bc055726855f2416e5b9ede631bd16d69b0366d425befb83048c600b492b" %}
+
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: premailer-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/{{ pypi_url_route }}/{{ name }}-{{ version }}.tar.gz#md5={{ pypi_url_md5 }}
```
Comment (jakirkham) [inline recipes/premailer/meta.yaml]: Please replace with the following.

```
https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

File: recipes/premailer/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{% set name = "premailer" %}
+{% set version = "3.0.1" %}
+{% set sha256 = "4e71cc09ad1438f827d1070ffac54ceb3a6a07c995fa82cb34c1ef163adeb432" %}
+
+{% set pypi_url_md5 = "aa15b483236eab47c3db5e4e5eb77cf0" %}
+{% set pypi_url_route = "73/2e/bc055726855f2416e5b9ede631bd16d69b0366d425befb83048c600b492b" %}
```
Comment (jakirkham) [inline recipes/premailer/meta.yaml]: Once the URL is changed. Both of these can be dropped.

File: recipes/cssutils/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+{% set version = "1.0.1" %}
+{% set sha256 = "d8a18b2848ea1011750231f1dd64fe9053dbec1be0b37563c582561e7a529063" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: cssutils-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/c/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/cssutils/meta.yaml]: Please change `pypi.python.org` -> `pypi.io`.

Also, please change `/c/` -> `/{{ name[0] }}/`.

**Comment by vamega** [comment]
I'm not sure why this failed on Appveyor. The logs end somewhat abruptly.
I'll take a more close look this evening, but I suspect it might have been transient.

Is there a way to restart that job?

**Comment by jakirkham** [comment]
Restarted.

**Comment by vamega** [comment]
Looking like it's crashing because the `ProgramFiles` environment variable isn't set.
I'm not sure what's going on there to cause that.

Only happens in the last of the packages, i.e. `premailer`.
Any help debugging this would be greatly appreciated.

**Comment by jakirkham** [comment]
`conda-build-all` deployment issue ( https://github.com/conda-forge/conda-build-all-feedstock/issues/12 ) was fixed. So had restarted the CIs. They are now all passing.

File: recipes/cssselect/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+{% set version = "0.9.2" %}
+{% set sha256 = "76f090e40297b21fb57459b0ea73c8e46046feaf872909224d1fb861c1703ef5" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: cssselect-{{ version }}.tar.gz
+  url: https://github.com/scrapy/{{ name }}/archive/v{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/cssselect/meta.yaml]: Add the `name` seems too much here. Is it alright if we just make it `cssselect`.

File: recipes/cssselect/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{% set name = "cssselect" %}
+{% set version = "0.9.2" %}
+{% set sha256 = "76f090e40297b21fb57459b0ea73c8e46046feaf872909224d1fb861c1703ef5" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: cssselect-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/cssselect/meta.yaml]: This could just be `{{ cssselect }}-{{ version }}.tar.gz`.

File: recipes/cssselect/meta.yaml (hunk 6 above)
Comment (vamega) [inline recipes/cssselect/meta.yaml]: I'll change it, but any particular reason you want it this way?

File: recipes/cssselect/meta.yaml (hunk 7 above)
Comment (vamega) [inline recipes/cssselect/meta.yaml]: Will do.

**Comment by jakirkham** [comment]
Could you please rebase on `master`? I think we are running into issue ( https://github.com/SciTools/conda-build-all/issues/52 ).

File: recipes/cssselect/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/cssselect/meta.yaml]: The main purpose of using jinja templates to me is to provide an abstraction (that of course comes at some cost in clarity) in order to do either or both of these things.

1. Make it easy to update.
2. Be a building block for new recipes.

Version helps 1. Checksum has some small value by making it easier to update at the top. So we have become ok with it. Name doesn't matter for 1, but helps 2. However, having it in the URL the way it is now doesn't help 1 or 2. For 1 the URL has to be removed anyway. For 2 it doesn't provide any advantage that hard-coding doesn't already satisfy. As mentioned, it already makes that URL a little bit unclear, but as it doesn't offer anything in exchange it should be dropped.

**Comment by vamega** [comment]
I rebased on master, but it seems like I still ran into the issue.

**Comment by vamega** [comment]
I think this got closed by an accidental force push to my branch. Reopening.

File: recipes/premailer/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,43 @@
+{% set name = "premailer" %}
+{% set version = "3.0.1" %}
+{% set sha256 = "4e71cc09ad1438f827d1070ffac54ceb3a6a07c995fa82cb34c1ef163adeb432" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: premailer-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/premailer/meta.yaml]: Could be `{{ name }}-{{ version }}.tar.gz`.

File: recipes/cssutils/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,37 @@
+{% set name = "cssutils" %}
+{% set version = "1.0.1" %}
+{% set sha256 = "d8a18b2848ea1011750231f1dd64fe9053dbec1be0b37563c582561e7a529063" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: cssutils-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/cssutils/meta.yaml]: Could just be `{{ name }}-{{ version }}.tar.gz`.

**Comment by jakirkham** [comment]
Some things to address still, but they are easy.

**Comment by vamega** [comment]
Could me merge this if there are no other issues?

**Comment by msarahan** [comment]
Thanks @vamega

**Pull request merged**